sdk/python/feast/infra/offline_stores/snowflake_source.py

```diff
@@ -183,6 +179,7 @@ def to_proto(self) -> DataSourceProto:
             A DataSourceProto object.
         """
         data_source_proto = DataSourceProto(
+            name=self.name,
```

Comment:
Was this the main bug leading to duplicate data sources?

Reply:
yes this was the root cause, see https://github.com/feast-dev/feast/issues/2581#issuecomment-1107694083

Reply:
Good catch.